docs/source-pytorch/release_policy.rst

```diff
@@ -0,0 +1,41 @@
+.. _release_policy:
+
+Release Policy
+##############
+
+This document describes release policy we follow for PyTorch Lightning.
+
+Versioning
+**********
+
+A Lightning release number is in the format of ``MAJOR.MINOR.PATCH``.
+
+We release a new minor version every three months and bugfix release every week.
+
+Patch releases contain only bugfixes and introduce no breaking changes, so it is always recommended to update the package to the latest version within the minor version whenever possible.
+However, unlike `semantic versioning (SemVer) <https://semver.org/>`_, minor releases contain API changes, removals, potential backward-incompatible
+changes as well as new features, deprecations, and all previous bugfixes included in any bugfix release. With every release, we publish a changelog
+where we list additions, removals, deprecations, changed functionality and fixes.
+
+API Evolution
+*************
+
+Lightning's development is driven by research and best practices in a rapidly developing field of AI and machine
+learning. Change is inevitable and when it happens, the Lightning team is committed to minimizing user friction and
+maximizing ease of transition from one version to the next. We take backward compatibility and reproducibility very
+seriously.
+
+For API removal, renaming or other forms of backward-incompatible changes, the procedure is:
+
+#. A deprecation process is initiated at a minor version ``X``, producing warning messages at runtime and in the documentation.
+#. Calls to the deprecated API remain unchanged in their function during the deprecation phase.
+#. Two minor versions in the future at version ``X+2`` the breaking change takes effect.
+
+The "X+2" rule is a recommendation and not a strict requirement. Longer deprecation cycles may apply for some cases.
```

Comment:
> The "X+2" rule is a recommendation and not a strict requirement. Longer deprecation cycles may apply for some cases.

In what situation, it is likely that we don't follow this rule?

Also, I see there might be longer deprecation cycles such as for [these](https://github.com/Lightning-AI/lightning/blob/273a9ed8c14650c59b60c546fe53d93a4183d439/tests/tests_pytorch/deprecated_api/test_remove_2-0.py) but do we have **shorter** deprecations, too? If yes, we need to document it as well.

Reply:
Speaking from what I experienced of the past: Default is X+2 and on a case-by-case basis, deviations sometimes get discussed. 

Example for a shorter deprecation:
- Function is broken/unusable beyond repair, and gets deprecated or changed. Shorter period because the feature is not usable in the current version. (DDP2 was a concrete example)

Example for a longer deprecation:
- An API that is so core to the Lightning experience that it is hard and time-consuming to make the switch for everyone involved, user and developer (us): Trainer(gpus=x) -> Trainer(devices=x), Trainer.fit() -> Trainer.train() etc. The user gets into contact with these APIs in almost all use cases.